pg_bm25/src/parade_index/index.rs

```diff
@@ -361,9 +244,17 @@ impl ParadeIndex {
         )
     }
 
-    pub fn scan_state(&self, config: &SearchConfig) -> TantivyScanState {
-        self.reload();
-        TantivyScanState::new(self, config)
+    pub fn scan_state(&self, config: &SearchConfig) -> Result<TantivyScanState, ParadeIndexError> {
+        // Prepare to perform a search.
+        // In case this is happening in the same transaction as an index build or an insert,
+        // we want to commit first so that the most recent results appear.
+        let writer_client = self.writer_client.clone();
```

Comment:
Is it safe to clone `Arc`s? Why not just get a reference to it?

Reply:
The writer client needs to be passed to the xact_callback (end of transaction hook), which has similar requirements to passing it to another thread. You can't send a reference over these kinds of contexts.

That's the purpose of Arc/Rc. They reference-count their inner value, so you get sort of a mini garbage collection system that allows you pass values around threads, callbacks, etc. `clone()` on Arc/Rc is basically free, it just increments the reference count.